src/functions.php

```diff
@@ -276,49 +310,18 @@ function countFilters(array $filters): int
 }
 
 /**
- * Check a list of filters against a card
+ * Check a list of filters against the vcard properties CATEGOTIES and/or GROUPS
  *
- * @param stdClass $vcard
+ * @param mixed $vcard
  * @param array $filters
  * @return bool
  */
-function filtersMatch(stdClass $vcard, array $filters): bool
+function filtersMatch($vcard, array $filters): bool
 {
     foreach ($filters as $attribute => $values) {
-        if (isset($vcard->$attribute)) {
-            if (filterMatches($vcard->$attribute, $values)) {
-                return true;
-            }
-        }
-    }
-
-    return false;
-}
-
-/**
- * Check a filter against a single attribute
- *
- * @param mixed $attribute
- * @param mixed $filterValues
- * @return bool
- */
-function filterMatches($attribute, $filterValues): bool
-{
-    if (!is_array($filterValues)) {
-        $filterValues = [$filterValues];
-    }
-
-    foreach ($filterValues as $filter) {
-        if (is_array($attribute)) {
-            // check if any attribute matches
-            foreach ($attribute as $childAttribute) {
-                if ($childAttribute === $filter) {
-                    return true;
-                }
-            }
-        } else {
-            // check if simple attribute matches
-            if ($attribute === $filter) {
+        $param = strtoupper($attribute);
+        if (isset($vcard->$param)) {
+            if (array_intersect($vcard->$param->getParts(), $values)) {
```

Comment:
Short and easy: getParts() delivers always an array and $values are an array depending on example.config.php.
Makes filterMatches superflous.